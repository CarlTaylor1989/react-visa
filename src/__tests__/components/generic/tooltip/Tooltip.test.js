/* global toJson */
import React from 'react';
import {
  Tooltip,
  mapStateToProps
} from '../../../../components/generic/tooltip/Tooltip';

let mockProps;
beforeEach(() => {
  mockProps = {
    children: 'irrelevant',
    fieldName: 'generic.tooltips.settings.audio',
    enabled: false,
    textProps: {
      score: 3
    }
  };
});

test('renders a disabled Tooltip', () => {
  const wrapper = shallow(<Tooltip {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders an enabled Tooltip', () => {
  const wrapper = shallow(<Tooltip {...mockProps} enabled />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('cannot render a Tooltip', () => {
  const wrapper = shallow(<Tooltip {...mockProps} forceHidden />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('should mapStateToProps', () => {
  // setup
  const state = {
    settingsData: {
      tooltips: false
    }
  };
  const expected = {
    enabled: state.settingsData.tooltips
  };

  // function under test
  const props = mapStateToProps(state);

  // expectation
  expect(props).toEqual(expected);
});
