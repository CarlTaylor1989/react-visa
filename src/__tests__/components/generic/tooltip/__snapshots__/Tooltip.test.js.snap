// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cannot render a Tooltip 1`] = `
<Fragment>
  <ForwardRef(TippyWrapper)
    arrow={true}
    content={
      <Translate
        dangerousHTML={true}
        score={3}
        tag="span"
        value="generic.tooltips.settings.audio"
      />
    }
    delay={
      Array [
        500,
        null,
      ]
    }
    enabled={false}
    theme="visa"
    touch={false}
    trigger="mouseenter"
  >
    irrelevant
  </ForwardRef(TippyWrapper)>
</Fragment>
`;

exports[`renders a disabled Tooltip 1`] = `
<Fragment>
  <ForwardRef(TippyWrapper)
    arrow={true}
    content={
      <Translate
        dangerousHTML={true}
        score={3}
        tag="span"
        value="generic.tooltips.settings.audio"
      />
    }
    delay={
      Array [
        500,
        null,
      ]
    }
    enabled={false}
    theme="visa"
    touch={false}
    trigger="mouseenter"
  >
    irrelevant
  </ForwardRef(TippyWrapper)>
</Fragment>
`;

exports[`renders an enabled Tooltip 1`] = `
<Fragment>
  <ForwardRef(TippyWrapper)
    arrow={true}
    content={
      <Translate
        dangerousHTML={true}
        score={3}
        tag="span"
        value="generic.tooltips.settings.audio"
      />
    }
    delay={
      Array [
        500,
        null,
      ]
    }
    enabled={true}
    theme="visa"
    touch={false}
    trigger="mouseenter"
  >
    irrelevant
  </ForwardRef(TippyWrapper)>
</Fragment>
`;
